### ISSUE TITLE
Add support in update_media tasks for updating field types in addition to plain text

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#606]
The new `update_media` task introduced in #602 allows us to update plain text fields on media, but not other field types. We should be able to update all supported field types.

**Comment by dmer** [comment]
@mjordan I've got a use case where I need to batch update the "Media of" field on a bunch of medias - I'm guessing based on your comment in #602 that this field (entity reference) is not yet supported is that correct?

**Comment by mjordan** [comment]
That's correct. However, I plan on adding support for all field types to update_media tasks before the start of the new year.

**Comment by dmer** [comment]
Oh that's great news - pls let me know if I can help w/ testing!

**Comment by dmer** [comment]
Hi @mjordan just checking in on the ETA for adding support for all field types to update_media tasks. I have a project that will need this functionality and I'd like to sketch out a timeline for them that is realistic.
Thanks!

**Comment by mjordan** [comment]
Hi @dmer. I'm sorry but don't have a specific ETA for you since I don't know how much time it will take to do this. My plan is to reuse the approach used to update [all field types in nodes](https://github.com/mjordan/islandora_workbench/blob/main/workbench#L515-L548), but the `update_media` task was implemented in its own special way. I can take an initial stab at this this week for sure but that's just to figure out if the node approach will work.

**Comment by dmer** [comment]
Thanks @mjordan - that's really appreciated! Hit me up for testing when you need it and I'll make time.

**Comment by mjordan** [comment]
Sorry my capacity to work on workbench has been so limited this last month or so.

**Comment by mjordan** [comment]
@dmer `issue_606` branch is ready for initial testing. Thanks!

**Comment by mjordan** [comment]
What line does python report the `AttributeError` occurred on?

**Comment by dmer** [comment]
Here's the full trace:

"Update media" task started using config file update_media.yml.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/islandora/islandora_workbench/./workbench:2286 in <module>                                 │
│                                                                                                  │
│   2283 │   if config['task'] == 'create_terms':                                                  │
│   2284 │   │   create_terms()                                                                    │
│   2285 │   if config['task'] == 'update_media':                                                  │
│ ❱ 2286 │   │   update_media()                                                                    │
│   2287 │   if config['task'] == 'update_terms':                                                  │
│   2288 │   │   update_terms()                                                                    │
│   2289                                                                                           │
│                                                                                                  │
│ /home/islandora/islandora_workbench/./workbench:1061 in update_media                             │
│                                                                                                  │
│   1058 │   for row in csv_data:                                                                  │
│   1059 │   │   # Delete expired items from request_cache before processing a row.                │
│   1060 │   │   if config['enable_http_cache'] is True:                                           │
│ ❱ 1061 │   │   │   requests_cache.delete(expired=True)                                           │
│   1062 │   │                                                                                     │
│   1063 │   │   media_id = extract_media_id(config, row)  # Extract the media ID from the CSV ro  │
│   1064 │   │   if media_id is None:  # If the media ID is invalid, skip this row                 │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: module 'requests_cache' has no attribute 'delete'

**Comment by mjordan** [comment]
And you don't get that error during any other task?

**Comment by mjordan** [comment]
HI @dmer sorry to be a pest but I'd like to resolve this so I can move on to other issues. Thanks for confirming that the updates work. I'm a bit perplexed at the cache error since that code isn't new, and I would expect the error you encountered to be triggered within every task.

**Comment by dmer** [comment]
Hi @mjordan I'm now convinced that error was a result of my config missing a component - I don't get that error once I correctly set the cache configs and I don't get it in any other tasks either. Sorry to introduce a red-herring!
Big thanks for getting this media field update in place! 🙌

**Comment by mjordan** [comment]
Thanks, I'm glad you resolved it. I'll move on to completing the new `--check` stuff and merge. Thanks for testing.

**Comment by mjordan** [comment]
I have merged the issue_606 branch into main. Updating docs and adding integration tests still to come. I should have the docs done early tomorrow.

**Comment by mjordan** [comment]
Ah thanks. `--check` is always the last thing I complete when adding new functionality. I'll take a look and fix.

**Comment by mjordan** [comment]
I can't replicate the `--check` error. It must be the custom media type. There are a couple other issues open for this problem, or related ones, that I haven't gotten to. I'll see what I can find out.

**Comment by mjordan** [comment]
Can you share your config (sans host, user, and password) with me please?

**Comment by dmer** [comment]
Unfortunately that config is on a server that requires special VPN access and requires me to be in the office to connect. I'll be able to get that tomorrow and will post here.

**Comment by dmer** [comment]
Hi Mark - 

Here's the config I was using (minus the host and pw lines) - the logging bits I added as I was trying to troubleshoot the error from running check. When I ran it without check there were no issues or errors and the update worked exactly as I expected. Also as I mentioned, this site uses a unique custom media type which may well be the cause of this error. I'm not blocked by this right now but am happy to pursue the resolution - just let me know what I can do!

task: update_media
input_dir: input_data
input_csv: newspaper_test_update_media.csv
log_json: true
log_request_url: true
log_headers: true
log_response_body: true
published: true
standalone_media_url: true
strict_check: false
enable_http_cache: true
http_cache_storage: memory
http_cache_storage_expire_after: 1000

**Comment by mjordan** [comment]
Using custom media types is very buggy, but I don't see any configuration items in your config file that would register your custom media type. Did you add any?

If you have a minute, I'm collecting input over at #737 and I'd love you hear what your needs around custom media type are.

**Comment by dmer** [comment]
That config didn't have any items regarding the media type. In this case all I wanted was to update an entity reference field (field_media_of) and that seems to have worked in spite of the odd media type.

I have to get ready for another meeting now that may eat the rest of my day, but I'll put a reminder for myself to add some thoughts to that issue in the morning..
